File: clase05/polimorfismo3/bicicleta.cpp
#include "bicicleta.h"


int id_aux_bici = 0;

Bicicleta::Bicicleta(){
  
  this->estado_motor = false;
  this->marcha = 0;
  this->bicicleta_id = ++id_aux_bici;
}


Bicicleta::Bicicleta(string modelo){
  this->estado_motor = false;
  this->marcha = 0;
  this->bicicleta_id = ++id_aux_bici;
  
  this->modelo = modelo;  
}

Bicicleta::~Bicicleta(){
  cout<< "Bicicleta " << this->bicicleta_id << " Destruida" << endl;
}

void Bicicleta::encender(){  
  cout << "La bicicleta " << this->bicicleta_id << " ";
  cout << "No puede ser encendida" << endl;
}

void Bicicleta::aumentar_marcha(){
  cout << "No puede aumentar marcha" << endl;
}

void Bicicleta::acelerar(int v){
  if(this->velocidad < v){
    this->velocidad = v;
  }
  cout << "Esta pedaleando más rapido" << endl;
}

File: clase06/oop/Bicicleta.java
package clase06.oop;

public class Bicicleta extends Vehiculo {
            
    public Bicicleta(String modelo) {
        super();
        this.Modelo = modelo;
    }
    
    public void Encender() {               
        System.out.println("Soy una bici sin motor");
        System.out.println("No me puedo encender, pero si puedes encender las luces");
    };
    
    public int AumentarMarcha() {
        ++(this.Marcha);
        return this.Marcha;
    };
    
    public void Acelerar(int v) {
        if (this.Velocidad < v) {
            this.Velocidad = v;
        }
        System.out.println("Esta pedaleando más rápido, a una velocidad de " + this.Velocidad + " km/h");
    };
}
